tutorials/plugins/gdnative/gdnative-c-example.rst

```diff
@@ -322,172 +327,232 @@ The variant we return is destroyed automatically by Godot.
         return ret;
     }
 
-And that is the whole source code of our module.
+Strings are heap-allocated in Godot, so they have a destructor which frees the
+memory. Destructors are named ``godot_TYPENAME_destroy``. When a Variant gets
+created with a String, it references the String. That means that the original
+String can be "destroyed" to decrease the ref-count. If that does not happen the
+String memory will leak since the ref-count will never be zero and the memory
+never deallocated. The returned variant gets automatically destroyed by Godot.
+
+.. note::
+
+    In more complex operations it can be confusing the keep track of which value
+    needs to be deallocated and which does not. As a general rule: call
+    ``godot_TYPENAME_destroy`` when a C++ destructor would be called instead.
+    The String destructor would be called in C++ after the Variant was created,
+    so the same is necessary in C.
 
-If you add a blank ``.gdignore`` file to the src folder, Godot will not try to
-import the compiler-generated temporary files.
+The variant we return is destroyed automatically by Godot.
+
+And that is the whole source code of our module.
 
 Compiling
 ---------
 
 We now need to compile our source code. As mentioned our example project on
-GitHub contains a Scons configuration that does all the hard work for you but
+GitHub contains a SCons configuration that does all the hard work for you, but
 for our tutorial here we are going to call the compilers directly.
 
-Assuming you are sticking to the folder structure suggested above it is best to
-CD into the src subfolder in a terminal session and execute the commands from
-there. Make sure to create the bin folder before you proceed.
+Assuming you are sticking to the folder structure suggested above, it is best to
+open a terminal session in the ``src`` folder and execute the commands from
+there. Make sure to create the ``bin`` folder before you proceed.
 
 On Linux:
 
 .. code-block:: none
 
-    clang -std=c11 -fPIC -c -I/PATH/TO/GODOT/HEADERS simple.c -o simple.os
-    clang -shared simple.os -o ../bin/libsimple.so
+    gcc -std=c11 -fPIC -c -I../../godot_headers simple.c -o simple.os
+    gcc -shared simple.os -o ../bin/libsimple.so
 
 On macOS:
 
 .. code-block:: none
 
-    clang -std=c11 -fPIC -c -I/PATH/TO/GODOT/HEADERS simple.c -o simple.os -arch i386 -arch x86_64
+    clang -std=c11 -fPIC -c -I../../godot_headers simple.c -o simple.os -arch i386 -arch x86_64
     clang -dynamiclib simple.os -o ../bin/libsimple.dylib -arch i386 -arch x86_64
 
 On Windows:
 
 .. code-block:: none
 
-    cl /Fosimple.obj /c simple.c /nologo -EHsc -DNDEBUG /MD /I. /IC:\PATH\TO\GODOT\HEADERS
+    cl /Fosimple.obj /c simple.c /nologo -EHsc -DNDEBUG /MD /I. /I..\..\godot_headers
```

Comment:
I didn't check if that works on Windows, but given that there is `/I.`, I guess relative paths are fine?

Reply:
I've had problems with relative paths when you start placing source files in subfolders. Not important for the example but..

Reply:
Though when you reach that point you likely have a SCons setup or `Makefile` that takes care of properly setting include paths for the whole codebase, so I guess for the example it's simple to give a copy-paste ready command (assuming that users followed the recommended project setup).